Add Reshape if there is add_axis when exporting C2 concat

yinghai opened the pull request:
Summary: The semantics of C2 and ONNX Concat is a bit different. C2 concat accepts "add_axis" arg and will raise the dim if so. It's equivalent of attaching a Reshape after plain concat in ONNX.

Differential Revision: D13012867

houseroad commented on caffe2/onnx/onnx_exporter.cc:
```diff
@@ -644,27 +645,78 @@ ConvertedResult OnnxExporter::CreateLrnNodes(
   auto& node = nodes.back();
   bool explicit_axis = false;
-  for (const auto& a : def.arg()) {
+  int axis = 1;
```
I think default value is -1.

houseroad reviewed (commented):
Could you add some comment to elaborate what's the meaning of add_axis and the second output in the code? Otherwise, each time, we have to check the implementation in caffe2.

yinghai commented:
Addressed comments.

Commit:
Add Reshape if there is add_axis when exporting C2 concat (#13798)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/13798

The semantics of C2 and ONNX Concat is a bit different. C2 concat accepts "add_axis" arg and will raise the dim if so. It's equivalent of attaching a Reshape after plain concat in ONNX.

Reviewed By: rdzhabarov

Differential Revision: D13012867

fbshipit-source-id: dae10c85b19f1c765bc952ff4e71fe815e8c2536
